src/Console/CommandScanner.php

```diff
@@ -109,41 +109,40 @@ public function scanPlugin(string $plugin): array
      */
     protected function scanDir(string $path, string $namespace, string $prefix, array $hide): array
     {
-        $dir = new Folder($path);
-        $contents = $dir->read(true, true);
-        if (empty($contents[1])) {
+        if (!is_dir($path)) {
             return [];
         }
 
-        $classPattern = '/(Shell|Command)$/';
+        $classPattern = '/(Shell|Command)\.php$/';
+        $fs = new Filesystem();
+        $files = $fs->find($path, $classPattern);
+
         $shells = [];
-        foreach ($contents[1] as $file) {
-            if (substr($file, -4) !== '.php') {
-                continue;
-            }
-            $shell = substr($file, 0, -4);
-            if (!preg_match($classPattern, $shell)) {
-                continue;
-            }
+        foreach ($files as $fileInfo) {
+            $file = $fileInfo->getFilename();
 
-            $name = Inflector::underscore(preg_replace($classPattern, '', $shell));
+            $name = Inflector::underscore(preg_replace($classPattern, '', $file));
             if (in_array($name, $hide, true)) {
                 continue;
             }
 
-            $class = $namespace . $shell;
-            if (!is_subclass_of($class, Shell::class) && !is_subclass_of($class, Command::class)) {
+            $class = $namespace . $fileInfo->getBasename('.php');
+            if (!is_subclass_of($class, Shell::class)
+                && !is_subclass_of($class, Command::class)
+            ) {
                 continue;
             }
 
-            $shells[] = [
+            $shells[$path . $file] = [
                 'file' => $path . $file,
                 'fullName' => $prefix . $name,
                 'name' => $name,
                 'class' => $class,
             ];
         }
 
+        ksort($shells);
```

Comment:
Should you `array_values()` after this?

Reply:
We could but not really necessary, the keys are never used.